### PR TITLE
[0-size Tensor No.46、139、140、142、143、173、323] Add 0-size Tensor support for diff/nan_to_num

### PR DESCRIPTION
**Pull request by co63oc**
### PR Category

Execute Infrastructure

### PR Types

Improvements 

### Description

PaddleAPITest 测试错误是torch error，没有cuda error， 增加单测
46 paddle.diff  错误是torch error
CPU
image
GPU
image

139 paddle.nan_to_num
CPU
image
GPU
image

140 paddle.nanmean
CPU
image
GPU
image

142 paddle.nanquantile 错误为torch error
CPU
image
GPU
image
image

143 paddle.nansum
CPU
image
GPU
image

173 paddle.nn.functional.glu
CPU
image
GPU
image

### CODE REVIEW

**Comment by DanielSun11** [comment]
请验证一下 paddle.diff 在PaddleAPITest/report/0size_tensor_gpu/20250319/accuracy/coredump_error/error_config.txt中的config是否还有coredump。这些配置可能报torch error，请尝试下--paddle_only = True。  如果一些不合法的配置paddle.diff 没有检查出，请完善下参数检查

**Comment by co63oc** [comment]
> 请验证一下 paddle.diff 在PaddleAPITest/report/0size_tensor_gpu/20250319/accuracy/coredump_error/error_config.txt中的config是否还有coredump。这些配置可能报torch error，请尝试下--paddle_only = True。 如果一些不合法的配置paddle.diff 没有检查出，请完善下参数检查

image
验证通过，paddle.diff 是python端算子，调用concat, subtract，如果有问题其他算子测试时应该会显示

**Review by DanielSun11 — APPROVED** [review]
LGTM

**Review by wanghuancoder — APPROVED** [review]
LGTM
